Use ipld.ErrNotFound

hsanjuan opened the pull request:
Rationale: https://github.com/ipfs/go-ipfs/issues/7074

hsanjuan committed:
Use ipld.ErrNotFound

hsanjuan commented:
Depends on:

* https://github.com/ipfs/go-ipfs-blockstore/pull/48
* https://github.com/ipfs/go-merkledag/pull/54

Jorropo commented on filestore.go:
```diff
@@ -117,53 +118,46 @@ func (f *Filestore) AllKeysChan(ctx context.Context) (<-chan cid.Cid, error) {
-			return blockstore.ErrNotFound
+	if ipld.IsNotFound(err2) {
+		if ipld.IsNotFound(err1) {
+			return err1
 		}
+		// being here means err1 was nil and err2 NotFound.
 		return nil
-	default:
-		return err2
 	}
+
+	return err2
```
```suggestion
	if !ipld.IsNotFound(err2) {
		return err2
	}
	
	if ipld.IsNotFound(err1) {
		return err1
	}
	
	return nil
```

This functionally identical, but clearer to me (I really had issues understanding your code, the `return nil` mixed between both cases made it hard on my brain for some reason).

Jorropo reviewed (approved):
LGTM (I'll add the changements for my comment in a commit before I merge)

Jorropo committed:
refactor: follow the happy left practice in Filestore.DeleteBlock

Jorropo committed:
chore: update go-ipfs-blockstore and go-merkledag

The commits of ipfs/go-ipfs-blockstore and ipfs/go-merkledag has been moved from the PR one to the master merged one to avoid deps breaking when github is gonna GC the PR commit.
